### PR TITLE
Test faspsolver

### PR DESCRIPTION
**Pull request by arthbous**
Added tests of the FASPSOLVER. Looks good now

### CODE REVIEW

File: tests/faspsolver_tests/test_faspsolver.cpp [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+  solu1_ex.interpolate(ExactSolu);
+  dolfin::Function Error4(V1);
+  copy_dvector_to_Function(&Solu1_fasp,&Error4);
+  *(Error4.vector())-=*(solu1_ex.vector());
+  double error_norm4 = 0.0;
+  L2Error::Form_M L2error4(mesh1,Error4);
+  error_norm4 = assemble(L2error4);
+
+  // On the 300x300 mesh
+  UnitSquareMesh mesh3(250, 250);
```
Comment (maximilianmetti) [inline tests/faspsolver_tests/test_faspsolver.cpp]: This mesh is not 300x300.

File: tests/faspsolver_tests/test_faspsolver.cpp [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+
+
+
+
+  if ((error_norm1 < 1E-7) && (error_norm2 < 1E-7) && (error_norm1 < error_norm4) && (error_norm5 < error_norm1))
+  {
+    std::cout << "Success... the fasp solver is working\n";
+  }
+  else {
+    std::cout << "ERROR...the fasp solver is not working\n";
```
Comment (maximilianmetti) [inline tests/faspsolver_tests/test_faspsolver.cpp]: Can we make the error message more annoying looking?  If we're running 20 tests, we might overlook such a beautiful message.  I've been doing something like this:
```
printf("***\tERROR IN FASP SOLVER TEST\n");
    printf("***\n***\n***\n");
    printf("***\tFASP SOLVER TEST:\n");
    printf("***\tThe computed solution is wrong\n");
    printf("***\n***\n***\n");
    printf("***\tERROR IN FASP SOLVER TEST\n");
    fflush(stdout);
```

**Comment by maximilianmetti** [comment]
I took a look and it looks good.  Please address the two in-line comments I mentioned above and then I'll merge.

**Comment by maximilianmetti** [comment]
Also!  you need to update ``test_all.sh``

**Comment by arthbous** [comment]
Done. you can merge

**Comment by maximilianmetti** [comment]
working on this. It's not compiling anymore

**Pull request merged**